### PR TITLE
Database Setup w/ First Model and Test

### PR DESCRIPTION
**Pull request by wcthompson**
What: Created a simple postgresql database for movies, wrote a basic model and serialize function with tests.

Why: sfmovies needs a place to store it's data! Also a key part of my engineering training :)

Details:
- created database, user and movies table
- created initial migrations and seeds, added relevant node scripts
- created movie model with bookshelf.js and wrote my first test!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,12 @@ node_js:
   - 5
 env:
```
Comment (elnaz) [inline .travis.yml]: minor: If you prefer the blank line between each section of the config, probably update the first half of this to have spaces as well so it's consistent.

File: config/development.js [hunk 2]
```diff
@@ -1,5 +1,10 @@
 'use strict';
 
 module.exports = {
-  PORT: '3000'
+  PORT: '3000',
```
Comment (elnaz) [inline config/development.js]: https://github.com/pallavi-s/sfmovies/pull/2#discussion_r66490112

File: config/test.js [hunk 3]
```diff
@@ -1,5 +1,10 @@
 'use strict';
 
 module.exports = {
-  PORT: '3000'
+  PORT: '3000',
```
Comment (elnaz) [inline config/test.js]: https://github.com/pallavi-s/sfmovies/pull/2#discussion_r66490112

File: lib/libraries/knex.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+const Knex = require('knex');
+
+const DatabaseConfig = require('../../db');
+
+module.exports = Knex(DatabaseConfig);
+
```
Comment (elnaz) [inline lib/libraries/knex.js]: minor: extra new line

File: package.json [hunk 5]
```diff
@@ -4,14 +4,21 @@
   "description": "Will's API for movies filmed in San Francisco",
   "main": "lib/index.js",
   "scripts": {
+    "db:migrate": "knex migrate:latest --knexfile db/index.js",
+    "db:migrate:make": "knex migrate:make --knexfile db/index.js",
+    "db:reset": "if [ \"$NODE_ENV\" != \"production\" ]; then dropdb $(node -p \"require('./config').DB_NAME\"); npm run db:setup && npm run db:migrate; fi",
+    "db:rollback": "knex migrate:rollback --knexfile db/index.js",
+    "db:setup": "if [ \"$NODE_ENV\" != \"production\" ]; then createdb -O $(node -e \"var c = require('./config'); console.log(c.DB_USER, c.DB_NAME);\"); fi",
+    "db:setup:user": "if [ \"$NODE_ENV\" != \"production\" ]; then createuser $(node -p \"require('./config').DB_USER\"); fi",
+    "db:seed": "npm run db:reset && knex seed:run --knexfile db/index.js",
```
Comment (elnaz) [inline package.json]: minor: move db:seed above db:setup

File: package.json [hunk 6]
```diff
@@ -4,14 +4,21 @@
...
     "enforce": "istanbul check-coverage --statement 100 --branch 100 --function 100 --lines 100",
     "lint": "eslint .",
     "release:patch": "changelog -p && git add CHANGELOG.md && git commit -m 'updated CHANGELOG.md' && npm version patch && git push origin && git push origin --tags",
     "release:minor": "changelog -m && git add CHANGELOG.md && git commit -m 'updated CHANGELOG.md' && npm version minor && git push origin && git push origin --tags",
     "release:major": "changelog -M && git add CHANGELOG.md && git commit -m 'updated CHANGELOG.md' && npm version major && git push origin && git push origin --tags",
     "start": "nodemon --ignore test/",
     "test": "NODE_ENV=test istanbul cover _mocha -- test --require test/setup.js --recursive --timeout 30000",
-    "test:raw": "NODE_ENV=test mocha test --require test/setup.js --recursive --timeout 30000"    
+    "test:raw": "NODE_ENV=test mocha test --require test/setup.js --recursive --timeout 30000"
```
Comment (elnaz) [inline package.json]: 👍

**Comment by elnaz** [comment]
LGTM

File: db/seeds/data/movies.json [hunk 7]
```diff
@@ -0,0 +1,1100 @@
...
+{
+  "title": "Swingin' Along",
+  "release_year": 1961
+},
+{
+  "title": "Take the Money and Run",
+  "release_year": 1969
+},
+{
+  "title": "Terminator - Genisys",
```
Comment (charlesbjohnson) [inline db/seeds/data/movies.json]: giphy 1

**Comment by charlesbjohnson** [comment]
LGTM 🔥 👍 🔥

**Pull request merged**